records/generator.go

```diff
@@ -124,15 +127,19 @@ type EnumerationData struct {
 }
 
 // NewRecordGenerator returns a RecordGenerator that's been configured with a timeout.
-func NewRecordGenerator(httpClient *http.Client) *RecordGenerator {
-	enumData := EnumerationData{
-		Frameworks: []*EnumerableFramework{},
+func NewRecordGenerator(config Config) *RecordGenerator {
+	scheme, tlsClientConfig := httpcli.TLSConfig(config.MesosHTTPSOn, config.caPool)
+	return &RecordGenerator{
+		httpClient: httpcli.New(
+			config.iamConfig,
+			httpcli.Transport(&http.Transport{
+				DisableKeepAlives:   true, // Mesos master doesn't implement defensive HTTP
+				MaxIdleConnsPerHost: 2,
```

Comment:
If we're creating a new pool on every failover, any reason to not make this 1? or 0?